### DISCUSSION TITLE
Extract table row splitted across multiple pages

### DISCUSSION THREAD

**Discussion by jsanjay63** [jsvine/pdfplumber#768] [Ask for help with specific PDFs]
Hello, I have been using this library before and I am really amazed at how "easy" it is to extract data.
Recently, I came across a situation where the pdf table row(refer attached image) was split across multiple pages with a page break in between. I am trying to extract tabular data in a CSV from this pdf using pdfplumber, but am getting this data in separate rows in a csv. Basically, I would like to get this data in a single row. I know, with some post-processing, I could merge both rows. But, I am in need of a much more generic solution so that in cases, when the rows aren't splitted, the same solution could work. With pdfplumber, is there a way to identify if the row has a horizontal border or not? If this information is available, it could help in merging the rows otherwise I could skip merging.

In the attached image, grey color coded are the cell's content.

![tmp](https://user-images.githubusercontent.com/19145015/204080372-dc878b10-a907-4bbb-819a-c9153a96610b.png)

**Comment by samkit-jain** [comment]
Hi @jsanjay63 Appreciate your interest in the library. To solve for these cases, you would need to write a custom logic. You can do so by checking for any line/rect objects at the end of the last row on a page and if none, merge the next 2 rows. Or, if you know the type of data you are dealing with then you can use that as well. For example, combining rows until column A has a value, or similar.

**Reply by jsanjay63** [reply]
Appreciate your help on this. I am fine with writing custom logic. I tried checking the edges/lines/rects objects to identify the row split. But, unfortunately I couldn't find any hint to identify the row split. Is there any other property or object which could help resolve this? Thanks.

**Reply by samkit-jain** [reply]
Without the PDF, it is hard to assist. If you can provide the PDF (redacting any sensitive information), then I will be able to help better.

**Reply by jsanjay63** [reply]
Sure, PFA sample PDF. Thanks
[1234_Temp_ABC_1234.pdf](https://github.com/jsvine/pdfplumber/files/10153073/1234_Temp_ABC_1234.pdf)

**Reply by kathimohan** [reply]
@samkit-jain Is it possible to detect that kind by tables by changing any of this table properties - table_settings = {
    "vertical_strategy": "lines", 
    "horizontal_strategy": "lines",
    "explicit_vertical_lines": [],
    "explicit_horizontal_lines": [],
    "snap_tolerance": 3,
    "snap_x_tolerance": 3,
    "snap_y_tolerance": 3,
    "join_tolerance": 3,
    "join_x_tolerance": 3,
    "join_y_tolerance": 3,
    "edge_min_length": 3,
    "min_words_vertical": 3,
    "min_words_horizontal": 1,
    "keep_blank_chars": False,
    "text_tolerance": 3,
    "text_x_tolerance": 3,
    "text_y_tolerance": 3,
    "intersection_tolerance": 3,
    "intersection_x_tolerance": 3,
    "intersection_y_tolerance": 3,
}                                                     ---- Is there any doc, to understand each of these properties, If have can u pls share

**Reply by samkit-jain** [reply]
Hi @kathimohan Thanks for sharing the PDF. Upon looking at it, I think you can use the Appointment key as your terminating condition. If you use `text` for vertical and horizontal strategy, you will get a response like

| Date  | Type        | Speciality       | Care Team          | Description |
|-------|-------------|------------------|--------------------|-------------|
| 10/03 | Appointment | Vascular Surgery |                    |             |
| /2022 |             |                  | Temp, BNV MD       |             |
|       |             |                  | 000 WESTERN AVENUE |             |


In this, you can create a custom post processing logic where, all the rows until the `Type` column is filled, will be merged together. For example, you will essentially be converting 
| Date  | Type        | Speciality       | Care Team          | Description |
|-------|-------------|------------------|--------------------|-------------|
| 10/03 | Appointment | Vascular Surgery |                    |             |
| /2022 |             |                  | Temp, BNV MD       |             |
|       |             |                  | 000 WESTERN AVENUE |             |
| 10/03 | Appointment | Vascular Surgery |                    |             |
| /2022 |             |                  | Temp, BNV MD       |             |
|       |             |                  | 000 WESTERN AVENUE |             |

to

| Date       | Type        | Speciality       | Care Team                       | Description |
|------------|-------------|------------------|---------------------------------|-------------|
| 10/03/2022 | Appointment | Vascular Surgery | Temp, BNV MD 000 WESTERN AVENUE |             |
| 10/03/2022 | Appointment | Vascular Surgery | Temp, BNV MD 000 WESTERN AVENUE |             |

**Reply by samkit-jain** [reply]
You can also find all the different table settings with explanation at https://github.com/jsvine/pdfplumber#table-extraction-settings

**Reply by akshatmittal2223** [reply]
Hi @samkit-jain, using the post-processing logic based on the type column, if there is not any such key on basis we can merge the tables, is there any approach to do so. Please let me know if anything related available where I will be able to merge tables on multiple pages.

**Reply by samkit-jain** [reply]
Hi @akshatmittal2223 Could you please share the PDF and also what all you have tried? It is hard to assess without seeing the PDF. Please remove any sensitive information from it before sharing.

**Reply by akshatmittal2223** [reply]
@samkit-jain I have attached the screenshot from the pdf for reference. In my case, the columns are not fixed, and in single pdf I have almost 15 tables that too with different headers. I have used PyMuPDF library also for extracting the data.
![image](https://github.com/user-attachments/assets/3486cec0-1607-4ab2-b894-fccf1c21be3d)

**Reply by samkit-jain** [reply]
Post-processing logic that you can try
1. Absence of any text between the last table on the current page and the first table on the next page.
2. Coordinates of the vertical line separators being exactly same.

**Comment by kathimohan** [comment]
@samkit-jain Is it possible to combine two pdf pages and combine as one (opposite to page.crop functionality) and then extract tables from it

**Reply by samkit-jain** [reply]
I have never tried it and don't think it is possible out of the box. You can experiment and see if it is something that is doable or not. If it is, don't forget to raise a PR for it to be added to the FAQ section as it can be helpful for others as well.

**Comment by sahlebrahim** [comment]
i am also going through this situation, would love to know if you find any workaround. thanks
